File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/analysis/own-tests-project/src/surveycrawler/crawler_run.py
class CrawlerRun:

    # command processing

    def process_command(self, a_crawler, a_command):
        if a_crawler.is_start_guarded_run_command(a_command):
            self.start_guarded_run(a_crawler)
            return
        if a_crawler.is_finish_guarded_run_command(a_command):
            self.finish_guarded_run(a_crawler)
            return

        a_crawler.process_movement_command(a_command)

    def start_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def finish_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def assert_is_finished(self, a_crawler):
        raise NotImplementedError()

    # moving

    def slide_crawler_towards(self, a_crawler, a_direction, a_silt):
        raise NotImplementedError()

    # undoing

    def register_movement(self, a_movement):
        raise NotImplementedError()

    def undo_movements(self, a_crawler):
        raise NotImplementedError()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-210827-1958/output/project/src/main/java/surveycrawler/CrawlerRun.java
package surveycrawler;

public abstract class CrawlerRun {

    // movement registration

    public abstract void registerMovement(CrawlerMovement aMovement);

    // moving

    public abstract void slideOnSilt(SurveyCrawler aCrawler, Point aDirection);

    // guarded run

    public abstract CrawlerRun startGuardedRun(SurveyCrawler aCrawler);

    public abstract CrawlerRun endGuardedRun(SurveyCrawler aCrawler);

    // command processing

    public abstract void finish(SurveyCrawler aCrawler);

    public abstract void undoOn(SurveyCrawler aCrawler);
}
